src/adaptors/tcp_adaptor.c

```diff
@@ -982,13 +982,20 @@ static qdr_tcp_connection_t *qdr_tcp_connection_ingress(qd_tcp_listener_t* liste
     tc->bridge->connections_opened +=1;
     UNLOCK(tc->bridge->stats_lock);
 
+    LOCK(tc->activation_lock);
```

Comment:
Can you move this further down in the code?  Like right before calling pn_listener_raw_accept?  There's no need to lock at this point, and by calling it here we now nest locks with the log lock, which we really don't want to do.